klient: panic after startup

Happened on klient startup right after kd install.

Built from current master (8ebafd046af8cde78f7ef89269fe1d6c719de95c).

```
2016-12-20 19:11:28 [klient] INFO     [remote][restoreMounts queue] Remounted successfully.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x23821c]

goroutine 13 [running]:
panic(0xd4a700, 0xc42000c0e0)
	/Users/rjeczalik/go/src/runtime/panic.go:500 +0x1a1
koding/klient/machine.(*AddrBook).Latest(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/addr.go:123 +0x6c
koding/klient/machine/machinegroup/addresses.(*Addresses).Latest(0xc42037d560, 0xc42037d620, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/machinegroup/addresses/addresses.go:58 +0x128
koding/klient/machine/machinegroup/addresses.(*Cached).Latest(0xc42037d580, 0xc42037d620, 0x18, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/machinegroup/addresses/cached.go:62 +0xa6
koding/klient/machine/machinegroup.(*Group).dynamicAddr.func1(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xc420486f18, ...)
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/machinegroup/machinegroup.go:175 +0x9f
koding/klient/machine.(*DynamicClient).cron(0xc42012b790)
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/dynamic.go:171 +0x197
created by koding/klient/machine.NewDynamicClient
	/Users/rjeczalik/src/github.com/koding/koding/go/src/koding/klient/machine/dynamic.go:104 +0x271
```

--- a/go/src/koding/klient/machine/machinegroup/addresses/cached.go
+++ b/go/src/koding/klient/machine/machinegroup/addresses/cached.go
@@ -29,6 +29,13 @@ func NewCached(st storage.ValueInterface) (*Cached, error) {
 		return nil, err
 	}
 
+	// Drop inconsistent data.
+	for id, ab := range c.addresses.m {
+		if ab == nil {
+			delete(c.addresses.m, id)
+		}
+	}
+
 	return c, nil
 }
 

--- a/go/src/koding/klient/machine/machinegroup/aliases/cached.go
+++ b/go/src/koding/klient/machine/machinegroup/aliases/cached.go
@@ -29,6 +29,13 @@ func NewCached(st storage.ValueInterface) (*Cached, error) {
 		return nil, err
 	}
 
+	// Drop inconsistent data.
+	for id, alias := range c.aliases.m {
+		if alias == "" {
+			delete(c.aliases.m, id)
+		}
+	}
+
 	return c, nil
 }
 

--- a/go/src/koding/klient/machine/machinegroup/idset/idset.go
+++ b/go/src/koding/klient/machine/machinegroup/idset/idset.go
@@ -26,14 +26,14 @@ func Union(a, b machine.IDSlice) machine.IDSlice {
 // Intersection returns a set that contains all elements of A that also belong
 // to B, but no other elements.
 func Intersection(a, b machine.IDSlice) machine.IDSlice {
-	aset := make(map[machine.ID]bool)
+	aset := make(map[machine.ID]struct{})
 	for i := range a {
-		aset[a[i]] = true
+		aset[a[i]] = struct{}{}
 	}
 
 	res := make(machine.IDSlice, 0, len(a))
 	for i := range b {
-		if aset[b[i]] {
+		if _, ok := aset[b[i]]; ok {
 			res = append(res, b[i])
 		}
 	}

